src/quickgui/qgsquickutils.h

```diff
@@ -57,27 +60,68 @@ class QUICK_EXPORT QgsQuickUtils: public QObject
   public:
     //! Create new utilities
     QgsQuickUtils( QObject *parent = nullptr );
-    //! dtor
+    //! Destructor
     ~QgsQuickUtils() = default;
 
     //! \copydoc QgsQuickUtils::dp
     qreal screenDensity() const;
 
     /**
-      * Calculate the distance in meter representing baseLengthPixels pixels on the screen based on the current map settings.
+      * Creates crs from epsg code in QML
       */
-    Q_INVOKABLE double screenUnitsToMeters( QgsQuickMapSettings *mapSettings, int baseLengthPixels ) const;
+    Q_INVOKABLE static QgsCoordinateReferenceSystem coordinateReferenceSystemFromEpsgId( long epsg );
+
+    /**
+      * Creates QgsPointXY in QML
+      */
+    Q_INVOKABLE QgsPointXY pointXYFactory( double x, double y ) const;
+
+    /**
+      * Creates QgsPoint in QML
+      */
+    Q_INVOKABLE QgsPoint pointFactory( double x, double y ) const;
+
+    /**
+      * Converts QGeoCoordinate to QgsPoint
+      */
+    Q_INVOKABLE QgsPoint coordinateToPoint( const QGeoCoordinate &coor ) const;
+
+    /**
+      * Transforms point between different crs from QML
+      */
+    Q_INVOKABLE static QgsPointXY transformPoint( const QgsCoordinateReferenceSystem &srcCrs,
+        const QgsCoordinateReferenceSystem &destCrs,
+        const QgsCoordinateTransformContext &context,
+        const QgsPointXY &srcPoint );
+
+    /**
+      * Calculates the distance in meter representing baseLengthPixels pixels on the screen based on the current map settings.
+      */
+    Q_INVOKABLE static double screenUnitsToMeters( QgsQuickMapSettings *mapSettings, int baseLengthPixels );
 
     //! Log message in QgsMessageLog
     Q_INVOKABLE void logMessage( const QString &message,
                                  const QString &tag = QString( "QgsQuick" ),
                                  Qgis::MessageLevel level = Qgis::Warning );
 
     /**
-     * Returns a string with information about screen size and resolution
-     *
-     * Useful to log for debugging of graphical problems on various display sizes
-     */
+      * Returns QUrl to image from libary's /images folder.
+      */
+    Q_INVOKABLE QUrl getThemeIcon( const QString &name );
+
+    /**
+      * Converts point to string with given decimals (default decimals = 3),
+      * e.g. -2.234521, 34.4444421 -> -2.234, 34.444
+      */
+    Q_INVOKABLE static QString qgsPointToString( const QgsPoint &point, int decimals = 3 );
+
+    /**
+      * Converts distance in meters to human readable length with given decimals (default decimals = 1),
+      *  e.g. 1222.234 m -> 1.2 km
+      */
+    Q_INVOKABLE static QString distanceToString( qreal dist, int decimals = 1 );
```

Comment:
This should be using QgsUnitTypes

Reply:
I am not sure if I understand how to use QgsUnitTypes? This function should automatically resolve which unit types to use based on the input.

Reply:
I would imagine this as a parameter, but I'm open to black magic ;)